app/src/main/java/pm/gnosis/heimdall/helpers/SignatureStore.kt

```diff
@@ -0,0 +1,83 @@
+package pm.gnosis.heimdall.helpers
+
+import android.content.Context
+import io.reactivex.*
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.accounts.base.models.Signature
+import pm.gnosis.heimdall.common.di.ApplicationContext
+import pm.gnosis.heimdall.data.repositories.TransactionRepository
+import pm.gnosis.heimdall.ui.exceptions.SimpleLocalizedException
+import pm.gnosis.models.Transaction
+import java.math.BigInteger
+import java.util.concurrent.CopyOnWriteArraySet
+import javax.inject.Inject
+
+class SimpleSignatureStore @Inject constructor(
+        @ApplicationContext private val context: Context
+) : SignatureStore, ObservableOnSubscribe<Map<BigInteger, Signature>>, SingleOnSubscribe<Map<BigInteger, Signature>> {
```

Comment:
Why is this an `ObservableOnSubscribe`?

Reply:
because that's how you can create an observable with `Observable.create` and I want it to be possible to observe this store.